### ISSUE TITLE
Question about your code

### ISSUE THREAD

**Issue by han-yeol** [JinY97/GCTNet#1]
Hi

    def forward(self, x: Tensor, mask: Tensor = None) -> Tensor:
        queries = rearrange(self.queries(x), "b t (h d) -> b h t d", h=self.num_heads)
        keys = rearrange(self.keys(x), "b t (h d) -> b h d t", h=self.num_heads)
        values = rearrange(self.values(x), "b t (h d) -> b h t d", h=self.num_heads)
        energy = torch.matmul(queries, keys)

        if mask is not None:
            fill_value = torch.finfo(torch.float32).min
            energy.mask_fill(~mask, fill_value)

        scaling = self.emb_size ** (1 / 2)
        att = F.softmax(energy / scaling, dim=-1)
        att = self.att_drop(att)

        **out = torch.einsum('bhal, bhlv -> bhav ', att, values)
        out = torch.matmul(att, values)**

        out = rearrange(out, "b h t d -> b t (h d)")
        out = self.projection(out)
        return out`

In your code, Matrix multiplication is executed in MHA block twice.
I want to know It is the right code.

Regards,

**Comment by JinY97** [comment]
Thanks for your question. It is what I employed during debugging. My primary focus was to comprehend the utilization of torch.einsum. Subsequently, it was commented out. And it does not have any impact on the outcome.

**Comment by JinY97** [comment]
Simply, both matrix multiplication results are the same.

**Comment by han-yeol** [comment]
> Simply, both matrix multiplication results are the same.

I understood that the operation results were the same, but I thought of the author's different intentions. Thank you for your answer.
